Documentation/platforms/aws.md

```diff
@@ -0,0 +1,77 @@
+
+## Capacity Planning
+
+As a critical building block for distributed systems it is crucial to perform adequate capacity planning in order to support the intended cluster workload. As a highly available and strongly consistent data store increasing the number of nodes in an etcd cluster will generally affect performance adversely. This makes sense intuitively, as more nodes means more members for the leader to coordinate state across. The most direct way to increase throughput and decrease latency of an etcd cluster is allocate more disk I/O, network I/O, CPU and memory to cluster members. In the event it is impossible to temporarily divert incoming requests to the cluster, scaling the EC2 instances which comprise the etcd cluster members one at a time may improve performance. It is, however, best to avoid bottlenecks through capacity planning.
+
+The etcd team has produced a [hardware recommendation guide](https://github.com/coreos/etcd/blob/master/Documentation/op-guide/hardware.md) which is very useful for “ballparking” how many nodes and what instance type are necessary for your cluster.
+
+AWS provides a service for creating groups of EC2 instances which are dynamically sized to match load on the instances. While running etcd cluster instances in an Auto Scaling Group ([ASG](http://docs.aws.amazon.com/autoscaling/latest/userguide/AutoScalingGroup.html)) is fine, using the ASG to dynamically change the size of the etcd cluster is not recommended. Auto Scaling Groups do provide a number of benefits besides cluster scaling which include:
+
+* distribution of EC2 instances across Availability Zones (AZs)
+* EC2 instance fail over across availability zones
+* consolidated monitoring and life cycle control of instances within an ASG
+
+The use of an ASG to create a [self healing etcd cluster](#self-healing) is one of the design considerations when deploying an etcd cluster to AWS.
+
+## Cluster Design
+
+The purpose of this section is to provide foundational guidance for deploying etcd on AWS. The discussion will be framed by the following three critical design criteria about the etcd cluster itself:
+
+* block device provider: limited to the tradeoffs between EBS or instance storage (InstanceStore)
+* cluster topology: how many nodes should make up an etcd cluster; should these nodes be distributed over multiple AZs
+* managing etcd members: creating a static cluster of EC2 instances or using an ASG.
+
+The intended cluster workload will is the primary driver of cluster design decisions. A configuration store for microservices may require different design considerations than a distributed lock service, a secrets store, or the Kubernetes control plane. Cluster design tradeoffs include considerations such as
+
+* availability
+* data safety
+* performance/throughput
+* self healing
+
+which must be matched to the expectations of the intended workload to be served by the etcd cluster.
```

Comment:
I think this line can be removed, it's already been said that the deciding factor is the workload